### PR TITLE
Feature/objects 712 v2 column definitions

### PR DESCRIPTION
**Pull request by andreassiegel**
Updated the `ObjectEntity` to match the v2 entity as well as the column definitions.

Also replaced the `columnDefinition` to be consistent, and to let Hibernate do the column definition work.

### CODE REVIEW

**Comment by andreassiegel** [comment]
Ah, wait... There is an issue with creating objects due to missing default values.

File: src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java [hunk 1]
```diff
@@ -24,43 +34,69 @@
...
     private String type;
 
     @NotNull
-    @Column(name = "accountuuid")
+    @Type(type = "uuid-binary")
+    @Column(name = "accountUuid", length = UUID_LENGTH, nullable = false, updatable = false)
     private UUID accountUrn;
 
     @CreatedDate
     @Column(name = "created", insertable = true, updatable = false)
```
Comment (ghost) [inline src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java]: name should be `createdTimestamp`

File: src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java [hunk 2]
```diff
@@ -24,43 +34,69 @@
...
-    @Temporal(TemporalType.TIMESTAMP)
-    private Date created;
+    private Long created;
```
Comment (ghost) [inline src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java:76]: What is the benefit of a `Long` over a `DateTime` other than consistency with the past?  The converters can convert this to/from anything we want.  I think a `DateTime` makes more sense here and in `lastModifiedTimestamp`.

File: src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java [hunk 3]
```diff
@@ -24,43 +34,69 @@
...
 
+    @Size(max = DESCRIPTION_LENGTH)
+    @Column(name = "description", length = DESCRIPTION_LENGTH)
     private String description;
 
-    @Column(name="activeflag")
+    @Basic
+    @NotNull
+    @Column(name="activeFlag", nullable = false)
     private Boolean activeFlag;
```
Comment (ghost) [inline src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java:100]: good Java naming calls this `isActive` in the class.  I would argue the boolean type negates the usefulness of the `flag` in the database as well.

```
@Column(name="active", nullable = false)
private Boolean isActive;
```

File: src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java [hunk 4]
```diff
@@ -24,43 +34,69 @@
...
     private String name;
 
+    @Size(max = DESCRIPTION_LENGTH)
+    @Column(name = "description", length = DESCRIPTION_LENGTH)
     private String description;
 
-    @Column(name="activeflag")
+    @Basic
+    @NotNull
+    @Column(name="activeFlag", nullable = false)
```
Comment (ghost) [inline src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java:99]: What type column does this become?  I think we should we make it `bit(1)` and default to `true`.

File: src/main/java/net/smartcosmos/dao/objects/impl/ObjectPersistenceService.java [hunk 5]
```diff
@@ -117,11 +113,11 @@ public ObjectResponse create(String accountUrn, ObjectCreate createObject) {
         }
 
         // findByExample doesn't deal with dates, so we have to do it ourselves
-        Date modifiedAfterDate = new Date(0);
+        Long modifiedAfterDate = null;
         boolean modifiedAfterDateSpecified = false;
```
Comment (ghost) [inline src/main/java/net/smartcosmos/dao/objects/impl/ObjectPersistenceService.java]: We can't be doing things like this.  There is no need to create and use a separate variable to represent the presence of something.  Initialize the `modifiedAfterDate` field to `null` as it is here and later check it for `!= null`.  One less variable.

File: src/main/java/net/smartcosmos/dao/objects/impl/ObjectPersistenceService.java [hunk 6]
```diff
@@ -133,11 +129,11 @@ public ObjectResponse create(String accountUrn, ObjectCreate createObject) {
         for (ObjectEntity singleResult : queryResult)
         {
             // created is set at object creation time, and lastModified is not
-            Date singleResultLastModified = singleResult.getLastModified();
+            Long singleResultLastModified = singleResult.getLastModified();
             if (singleResultLastModified == null){
                 singleResultLastModified = singleResult.getCreated();
             }
-            if(!modifiedAfterDateSpecified || singleResultLastModified.after(modifiedAfterDate))
```
Comment (ghost) [inline src/main/java/net/smartcosmos/dao/objects/impl/ObjectPersistenceService.java:171]: We cannot be doing this.  We need to make the database do the work by invoking the query with a comparison for the `lastUpdatedTimestamp`.  We should never retrieve results and iterate or inspect the return wen that can be most effectively done by the database.

File: src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java (hunk 2 above)
Comment (andreassiegel) [inline src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java:76]: For the future it makes more sense, right. However, staying with the `Long` allows for reuse of old databases.
If we go for a v2-like default implementation, it should IMO be as compatible as possible.

File: src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java (hunk 4 above)
Comment (andreassiegel) [inline src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java:99]: It's a `bit(1)` column and defaults to `true`, but in v2 the default value was `false`.

File: src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java (hunk 3 above)
Comment (andreassiegel) [inline src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java:100]: Lombok would call the getter `getIsActive()` then -- that's even more ugly. :/

Comment (ghost) [inline src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java:100]: bad call on Lombok then

Comment (andreassiegel) [inline src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java:100]: Lombok just creates getters like `isActive()` for `boolean` fields, but not for `Boolean`. Meh.
https://projectlombok.org/features/GetterSetter.html

Alternative: Override the Lombok getter: http://stackoverflow.com/questions/18139678/lombok-how-to-customise-getter-for-boolean-object-field

**Pull request merged**